src/app/recover-funds/[address]/RecoverFundsButton.tsx

```diff
@@ -0,0 +1,172 @@
+'use client';
+import React, { useState } from 'react';
+import { useMemo } from 'react';
+import {
+  Address,
+  getL2Network,
+  L1ToL2MessageGasEstimator,
+  L1TransactionReceipt,
+} from '@arbitrum/sdk';
+import { Inbox__factory } from '@arbitrum/sdk/dist/lib/abi/factories/Inbox__factory';
+import { getBaseFee } from '@arbitrum/sdk/dist/lib/utils/lib';
+import { useNetwork, useSigner } from 'wagmi';
+import { getProviderFromChainId, getTargetChainId } from '../../../lib';
+import { BigNumber } from 'ethers';
+
+function RecoverFundsButton({
+  address,
+  balanceToRecover,
+  chainID,
+}: {
+  address: string;
+  balanceToRecover: BigNumber;
+  chainID: number;
+}) {
+  const [message, setMessage] = useState('');
+  const [loading, setLoading] = useState(false);
+  const { chain } = useNetwork();
+  const { data: signer } = useSigner();
+
+  const redeemButton = useMemo(() => {
+    if (!signer) return null;
+
+    if (chain?.id !== 1 && chain?.id !== 5) {
+      return (
+        <div>Unknown L1 chain id. This chain is not supported by this tool</div>
+      );
+    }
+
+    if (chain?.id !== chainID) {
```

Comment:
I would add a condition to check the targetChainId from the chain.id, so we are sure that we can send the retryable to the targetChainId's chain from the current connected network.